source/sgp_mode/AnalysisTools.h

```diff
@@ -32,40 +32,73 @@
 
 // start of getNecessarySites methods
 
-/*
- *
- * Input: Takes in a cpu and the identifier for a specific task
+using Library = sgpl::OpLibrary<
+    sgpl::Nop<>,
+    // single argument math
+    inst::ShiftLeft, inst::ShiftRight, inst::Increment, inst::Decrement,
+    // biological operations
+    // no copy or alloc
+    inst::Reproduce, inst::PrivateIO, inst::SharedIO,
+    // double argument math
+    inst::Add, inst::Subtract, inst::Nand,
+    // Stack manipulation
+    inst::Push, inst::Pop, inst::SwapStack, inst::Swap,
+    // no h-search
+    inst::Donate, inst::JumpIfNEq, inst::JumpIfLess, inst::Reuptake,
+    // if-label doesn't make sense for SGP, same with *-head
+    // and set-flow but this is required
+    sgpl::global::Anchor>;
+
+using Spec = sgpl::Spec<Library, CPUState>;
+
+
+
+/**
  *
- * Output: Returns a boolean representing a program's ability to do a logic task
+ * Input: A cpu of an organism
  *
- *Purpose: To return whether or not the organism can perform the given task
+ * Output: A length 64 emp bitset which describes the phenotype of organism
+ * such that the ith 1 in the bitset marks the completion of task i.
  *
+ * Purpose: Get the phenotype of an organism
  *
  */
-
-bool ReturnTaskDone(size_t task_id, CPU org_cpu) {
+emp::BitSet<64> ReturnTasksDone(CPU org_cpu) {
   bool if_task_true = false;
   org_cpu.Reset();
   org_cpu.state.self_completed = {1, 1, 1, 1, 1, 1, 1, 1, 1};
 
-  org_cpu.RunCPUStep(emp::WorldPosition::invalid_id, 100);
+  org_cpu.RunCPUStep(emp::WorldPosition::invalid_id, 400);
+
+  return *org_cpu.state.used_resources;
+}
 
-  if (org_cpu.state.used_resources->Get(task_id)) {
-    if_task_true = true;
-  }
 
-  return if_task_true;
+/*
+ *
+ * Input: Takes in a cpu and the identifier for a specific task
+ *
+ * Output: a boolean representing a program's ability to do a logic task
+ *
+ * Purpose: To return whether or not the organism can perform the given task
+ *
+ *
+ */
+bool ReturnTaskDone(size_t task_id, CPU org_cpu) {
+  return ReturnTasksDone(org_cpu).Get(task_id);
 }
 
+
+
+
 /**
  *
- * Input: Takes in a CPU and the number identifying a given task
+ * Input:
  *
- * Output: Returns a vector representing the full genome, reduced to 1s and 0s to show either
- *that an instruction is necessary to complete the task, or not respectively
+ * Output:
  *
- *Purpose: Is to return a vector that acts as a reduced program representation
- *of the necessary code lines to complete the given task
+ *Purpose: Takes all the calculation methods and calls them in order of having a
+ *simplified way of getting an organism's Physical Modularity
```

Comment:
This documentation looks like it got put in the wrong place somehow, doesn't apply to the code below and removes documentation that did look correct